<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <title></title>
</head>
<body>

</body>
</html>
<script>
//try catch 进行浏览器的异常信息捕获
//    js报错后，之后的代码都不在执行

//17行代码执行报错，如果用try catch捕获了异常，不影响下边的代码继续执行；
// 如果try中的代码执行出错，会默认的去执行catch中的代码
/*    try{
        console.log(num)
    }catch(e){//形参必须要写，一般为e
        console.log(e.message)//e.message可以收集当前代码报错的原因
    }
    console.log("ok");*/


    try{
        //js code
    }catch(e){
//        如果代码报错 执行catch中的代码
    }finally{
//        一般不用，不管try中的代码是否报错，都要执行finally中的代码
    }
//    既想捕获到错误信息，又想不让下边代码继续执行

try{
    console.log(num)
}catch(e){//形参必须要写，一般为e
    //console.log(e.message);//e.message可以收集当前代码报错的原因,把其进行统计
//    手动在浏览器中抛出一条错误信息 ，终止代码执行
    throw new Error("当前网路繁忙，请稍后再试");
    new ReferenceError//引用错误
    new TypeError //类型错误
    new RangeError//范围错误
}
console.log("ok");
</script>